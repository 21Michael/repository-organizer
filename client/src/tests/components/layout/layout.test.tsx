import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import store from "../../../storeReduxToolkit/configStore";
import Layout from "../../../components/layout/layout";
import { createMemoryHistory } from "history";

export default describe("layout: ", () => {
  const history = createMemoryHistory();
  let container: Element;
  beforeEach(() => {
    container = render(
      <Provider store={store}>
        <BrowserRouter>
          <Layout />
        </BrowserRouter>
      </Provider>
    ).container;
  });
  afterAll(cleanup);

  test("Route: /note", () => {
    history.push("/note");
    expect(container.querySelector("main")).toMatchSnapshot();
  });

  test("Route: /", () => {
    history.push("/");
    expect(container.querySelector("main")).toMatchSnapshot();
  });

  test("Route: /repository/details", () => {
    history.push("/repository/details");
    expect(container.querySelector("main")).toMatchSnapshot();
  });

  test("Route: /signin", () => {
    history.push("/signin");
    expect(container.querySelector("main")).toMatchSnapshot();
  });

  test("Route: /signup", () => {
    history.push("/signup");
    expect(container.querySelector("main")).toMatchSnapshot();
  });
});
