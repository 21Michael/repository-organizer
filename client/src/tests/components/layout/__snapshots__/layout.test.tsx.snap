// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout:  Route: / 1`] = `
<main
  class="main"
>
  <div
    class="main__wrapper"
  >
    <div
      class="table__wrapper"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="tr"
          >
            <th
              class="th"
            >
              name
            </th>
            <th
              class="th"
            >
              description
            </th>
            <th
              class="th"
            >
              stars
            </th>
            <th
              class="th"
            >
              creator name
            </th>
            <th
              class="th"
            >
              created at
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        />
      </table>
    </div>
  </div>
</main>
`;

exports[`layout:  Route: /note 1`] = `
<main
  class="main"
>
  <div
    class="main__wrapper"
  >
    <div
      class="table__wrapper"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="tr"
          >
            <th
              class="th"
            >
              name
            </th>
            <th
              class="th"
            >
              description
            </th>
            <th
              class="th"
            >
              stars
            </th>
            <th
              class="th"
            >
              creator name
            </th>
            <th
              class="th"
            >
              created at
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        />
      </table>
    </div>
  </div>
</main>
`;

exports[`layout:  Route: /repository/details 1`] = `
<main
  class="main"
>
  <div
    class="main__wrapper"
  >
    <div
      class="table__wrapper"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="tr"
          >
            <th
              class="th"
            >
              name
            </th>
            <th
              class="th"
            >
              description
            </th>
            <th
              class="th"
            >
              stars
            </th>
            <th
              class="th"
            >
              creator name
            </th>
            <th
              class="th"
            >
              created at
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        />
      </table>
    </div>
  </div>
</main>
`;

exports[`layout:  Route: /signin 1`] = `
<main
  class="main"
>
  <div
    class="main__wrapper"
  >
    <div
      class="table__wrapper"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="tr"
          >
            <th
              class="th"
            >
              name
            </th>
            <th
              class="th"
            >
              description
            </th>
            <th
              class="th"
            >
              stars
            </th>
            <th
              class="th"
            >
              creator name
            </th>
            <th
              class="th"
            >
              created at
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        />
      </table>
    </div>
  </div>
</main>
`;

exports[`layout:  Route: /signup 1`] = `
<main
  class="main"
>
  <div
    class="main__wrapper"
  >
    <div
      class="table__wrapper"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="tr"
          >
            <th
              class="th"
            >
              name
            </th>
            <th
              class="th"
            >
              description
            </th>
            <th
              class="th"
            >
              stars
            </th>
            <th
              class="th"
            >
              creator name
            </th>
            <th
              class="th"
            >
              created at
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        />
      </table>
    </div>
  </div>
</main>
`;
